internal/cmd/stack/environment.go

```diff
@@ -73,6 +119,89 @@ func setVar(cliCtx *cli.Context) error {
 	return nil
 }
 
+func (e *listEnvCommand) listEnv(cliCtx *cli.Context) error {
+	outputFormat, err := cmd.GetOutputFormat(cliCtx)
+	if err != nil {
+		return err
+	}
+
+	stackID := cliCtx.String(flagStackID.Name)
+
+	var query listEnvQuery
+	variables := map[string]interface{}{
+		"stack": graphql.ID(stackID),
+	}
+
+	if err := authenticated.Client.Query(cliCtx.Context, &query, variables); err != nil {
+		return err
+	}
+
+	var outputs []listEnvElementOutput
+	for _, attachedContext := range query.Stack.AttachedContexts {
+		for _, config := range attachedContext.Config {
+			outputs = append(outputs, config.toConfigElementOutput(attachedContext.ContextName))
+		}
+	}
+	for _, config := range query.Stack.RuntimeConfig {
+		var contextName *string
+		if config.Context != nil {
+			contextName = &config.Context.ContextName
+		}
+		outputs = append(outputs, config.Element.toConfigElementOutput(contextName))
+	}
+
+	switch outputFormat {
+	case cmd.OutputFormatTable:
+		return e.showOutputsTable(outputs)
+	case cmd.OutputFormatJSON:
+		return e.showOutputsJSON(outputs)
+	}
+
+	return fmt.Errorf("unknown output format: %v", outputFormat)
+}
+
+func (e *listEnvCommand) showOutputsTable(outputs []listEnvElementOutput) error {
+	tableData := [][]string{{"Name", "Type", "Value", "File Mode"}}
+	for _, output := range outputs {
+		var row []string
+
+		row = append(row, output.Name)
+		row = append(row, output.Type)
+		if output.Value != nil {
```

Comment:
I think we might need special handling for mounted files here. In the case of mounted files, the value is the base64-encoded content of the file. Depending on the file this could be huge.

I've got a couple of ideas:

1. (the simple option) just don't show the value for files in the table output (but still include it in the json output).
2. (the more complex option) decode the value if present, and only show the first X characters.

Personally I'd just go for option 1 and not bother showing the file content in the table output format.

Reply:
Okay. I'm now decoding files and will apply special formatting formatting for them. I'm now replacing newlines with spaces and I'm trimming the length. In a future release you may want to trim all long columns in a table.

Reply:
Ok - sounds good.